### PR TITLE
HK: adds Godtuner for GH Flower Quest Goal

### PR DESCRIPTION
**Pull request by qwint**
## What is this fixing or adding?
apparently this is required? and noone knew?

## How was this tested?
generated once and didn't play it

### CODE REVIEW

File: worlds/hk/__init__.py [hunk 1]
```diff
@@ -659,6 +659,8 @@ class HKItem(Item):
     def __init__(self, name, advancement, code, type: str, player: int = None):
         if name == "Mimic_Grub":
             classification = ItemClassification.trap
+        elif name == "Godtuner":
```
Comment (BadMagic100) [inline worlds/hk/__init__.py:662]: I believe this should not be needed because godtuner is in the keys pool which would make it be progression inherently.

Comment (qwint) [reply worlds/hk/__init__.py:662]: keys pool isn't called out in that init, and if they are not called out they fall back into presence in ExtractedData.py logic_items
reason it happens for world sense is the Dreamer group is called out to set to prog skip balancing

Comment (qwint) [reply worlds/hk/__init__.py:662]: confirmed with a python shell
```
import Generate
multi = Generate.main()
x = [item for item in multi.itempool if item.name =="Godtuner"][0]
x.classification
```
was filler on main, prog on this branch

Comment (BadMagic100) [reply worlds/hk/__init__.py:662]: Do simple keys also not count as prog then? Because that would be a problem also

Comment (qwint) [reply worlds/hk/__init__.py:662]: ah, 'advancement' is also set if the item is in item_effects which simple_key is, didn't see that at first but still no mention of godtuner in there either

Comment (BadMagic100) [reply worlds/hk/__init__.py:662]: Interesting, guess this is fine

**Review by BadMagic100 — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**